App/Sagas/TextileSagas.js

```diff
@@ -239,9 +239,26 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (camera === undefined || camera.processed === undefined) {
+      for (const uri of allPhotos.map((p) => p.uri)) {
+        yield put(TextileActions.newImage(uri))
+      }
+      allPhotos = []
+    }
+
+    const newPhotos = allPhotos.filter((photo) => {
+      return processed.indexOf(photo.uri) === -1
```

Comment:
I think we can optimize this big time. This is like O(n^2) since you have to iterate through `processed` for every `photo`. Check out https://github.com/textileio/textile-mobile/blob/master/App/Containers/TextilePhotos.js#L36. The basic idea is to convert the array of `processed` to an Object with a key for every item in `processed`. That way you can just do a direct lookup like `if (processedObject[photo.uri]) ...` and things are O(n).